com.unity.render-pipelines.core/Editor/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,181 @@
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+namespace UnityEditor.Rendering
+{
+	/// <summary>
+	/// Material upgrader and relevant utilities for SpeedTree8.
+	/// </summary>
+	public class SpeedTree8MaterialUpgrader : MaterialUpgrader
+    {
+        private enum WindQuality
+        {
+            None = 0,
+            Fastest,
+            Fast,
+            Better,
+            Best,
+            Palm,
+            Count
+        }
+
+        private static string[] WindQualityString =
+        {
+            "_WINDQUALITY_NONE",
+            "_WINDQUALITY_FASTEST",
+            "_WINDQUALITY_FAST",
+            "_WINDQUALITY_BETTER",
+            "_WINDQUALITY_BEST",
+            "_WINDQUALITY_PALM"
+        };
+
+        /// <summary>
+        /// Creates a material upgrader with only the renames in common between HD and Universal.
```

Comment:
Commenting on this as it appears to be API doc.
Not clear what "only" means here. Does it mean that renaming materials is the only thing the material upgrader does the same way in both HDRP and URP?

Reply:
If there's a venn diagram of renames that need to happen, it goes like this:
(properties only in HD ( properties in both ) properties only in universal)
"only" here refers to the intersection of the venn diagram. There's probably a better way of writing this though

Reply:
Changed to:
"Creates a material upgrader that handles the property renames that HD and Universal have in common."

Reply:
Changed again, and integrated all your feedback:

        /// <summary>
        /// Creates a material upgrader that handles the property renames that HD and Universal have in common when upgrading
        /// from the built-in SpeedTree8 shader.
        /// </summary>
        /// <param name="sourceShaderName">Original SpeedTree8 shader name.</param>
        /// <param name="destShaderName">New SpeedTree8 shader name.</param>
        /// <param name="finalizer">A delegate that postprocesses the material for the render pipeline in use.</param>

Reply:
Looks good to me. Thanks!